spec/models/loan_health_check_spec.rb

```diff
@@ -0,0 +1,181 @@
+          end
+          context '79% of questions answered' do
+            before do
+              allow(check).to receive(:loan).and_return(loan)
+              allow(loan).to receive(:criteria).and_return(instance_double(LoanResponseSet, progress_pct: 79))
+              check.recalculate
+            end
+
+            it { is_expected.to include :incomplete_loan_questions }
+          end
+          context '80% of questions answered' do
+            before do
+              allow(check).to receive(:loan).and_return(loan)
+              allow(loan).to receive(:criteria).and_return(instance_double(LoanResponseSet, progress_pct: 80))
+              check.recalculate
+            end
+
+            it { is_expected.to_not include :incomplete_loan_questions }
+          end
+        end
+        context 'and no late steps' do
+          let(:loan) { create(:loan, :prospective, :with_open_project_step) }
+
+          it { is_expected.to_not include :late_steps }
+        end
+        context 'and no end date' do
+          let(:loan) { create(:loan, :prospective, end_date: nil) }
+
+          it { is_expected.to_not include :late_steps }
+        end
+        context 'and a step more than one day late' do
+          let(:loan) { create(:loan, :prospective, :with_past_due_project_step) }
+
+          it { is_expected.to include :late_steps }
+        end
+
+        context 'created 15 days ago' do
+          context 'with only 14 steps' do
+            let(:loan) { create(:loan, :prospective, :with_a_number_of_recent_project_steps, step_count: 14, signing_date: 16.days.ago, end_date: 1.day.ago) }
+
+            it { is_expected.to include :sporadic_loan_updates }
+          end
+          context 'with 15 steps' do
+            let(:loan) { create(:loan, :prospective, :with_a_number_of_recent_project_steps, step_count: 15, signing_date: 16.days.ago, end_date: 1.day.ago) }
+
+            it { is_expected.to_not include :sporadic_loan_updates }
+          end
+        end
+
+        context 'created 30+ days ago' do
+          context 'with only 29 recent steps' do
+            let(:loan) { create(:loan, :prospective, :with_a_number_of_recent_project_steps, step_count: 29) }
+
+            it 'has found 29 steps' do
+              expect(loan.timeline_entries.merge(ProjectStep.recent).count).to eq 29
+            end
+            it { is_expected.to include :sporadic_loan_updates }
+          end
+          context 'with only 29 recent steps and 5 old steps' do
+            let(:loan) do
+              create(:loan,
+                :prospective,
+                :with_a_number_of_recent_project_steps,
+                :with_a_number_of_old_project_steps,
+                step_count: 29,
+                old_step_count: 5)
+            end
+
+            it 'has found 29 steps' do
+              expect(loan.timeline_entries.merge(ProjectStep.recent).count).to eq 29
+            end
+            it { is_expected.to include :sporadic_loan_updates }
```

Comment:
I thought we had realized this was the wrong rule? Wasn't it just that there needs to be at least one step in each future 30 day period until the end date or something? Did you discuss with Fuzzy?